app.js

```diff
@@ -218,13 +241,13 @@ function updateDevices(hubSlug){
     harmonyHubClient.getAvailableCommands().then(function(commands) {
       foundDevices = {}
       commands.device.some(function(device) {
-        deviceCommands = {}
-        device.controlGroup.some(function(group) {
-          group.function.some(function(func) {
-            deviceCommands[parameterize(func.label)] = {name: func.name, label: func.label, action:func.action.replace(/\:/g, '::')}
-          })
-        })
-        foundDevices[device.id] = {id: device.id, slug: parameterize(device.label), label:device.label, commands:deviceCommands}
+        deviceCommands = getCommandsFromControlGroup(device.controlGroup)
+        foundDevices[device.id] = {id: device.id, slug: parameterize(device.label), label:device.label}
+        Object.defineProperty(foundDevices[device.id], "commands", {
```

Comment:
`enumerable: false` is the key here. It keeps the commands from showing up stringify the object.